smokey/src/main/java/frc/robot/commands/DriveRampUpSpeedCommand.java

```diff
@@ -0,0 +1,64 @@
+// ************************************************************
+// Bishop Blanchet Robotics
+// Home of the Cybears
+// FRC - Charged Up - 2023
+// File: DriveRampUpSpeedCommand.java
+// Intent: Forms a command to ramp up the drivetrain subsystem speed until it reaches maximum.
+// ************************************************************
+
+// ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ 
+
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.DrivetrainSubsystem;
+import edu.wpi.first.wpilibj.Timer;
+
+public class DriveRampUpSpeedCommand extends CommandBase {
+
+    private DrivetrainSubsystem drivetrainSub;
+    private Timer timer = new Timer();
+    private final double powerIncrementDelaySeconds = 0.5;
+
+
+    public DriveRampUpSpeedCommand(DrivetrainSubsystem drivetrainSub) {
+        this.drivetrainSub = drivetrainSub;
+        // DO NOT declare drivetrainSub as requirement.  
+        // We need default drive commands to keep working while this command runs
+    }
+
+    /**
+     * Override of the initialize method on CommandBase - called on start of command
+     */
+    @Override
+    public void initialize() {
+        drivetrainSub.incrementPowerReductionFactor();
+        timer.reset();
+        timer.start();
+    }
+
+    /**
+     * Override of the execute method on CommandBase - Called every time the scheduler runs while the command is scheduled.
+     */
+    @Override
+    public void execute() {
+        if (timer.hasElapsed(this.powerIncrementDelaySeconds)) {
+            drivetrainSub.incrementPowerReductionFactor();
```

Comment:
Similar comment as before, seems we increment 1x and then 0.5 seconds later we do it again.

Reply:
Yes.  We increment every 0.5s until we reach full speed.  That's so that we don't go from 20% speed to full speed so abruptly.  This implements a form of acceleration control.